dcrsqlite/apisource.go

```diff
@@ -320,12 +317,83 @@ func (db *wiredDB) GetRawTransactionWithPrevOutAddresses(txid string) (*apitypes
 
 	msgTx := txhelpers.MsgTxFromHex(txhex)
 	if msgTx == nil {
-		return tx, prevOutAddresses, "", nil
+		return &apitypes.ExplorerTxData{
+			Tx:       tx,
+			VinAddrs: prevOutAddresses,
+		}
 	}
 
 	txType := txhelpers.DetermineTxTypeString(msgTx)
+	var txFee, txFeeRate float64
+	if txType == "Ticket" || (txType == "Regular" && len(tx.Vin[0].Coinbase) < 5) {
+		txFee = txhelpers.TxFee(msgTx)
+		txFeeRate = txhelpers.TxFeeRate(msgTx)
+	} else {
+		txFee = 0.0
+		txFeeRate = 0.0
+	}
+
+	vinfo := new(apitypes.VoteInfo)
+	if stake.DetermineTxType(msgTx) == stake.TxTypeSSGen {
+		validation, version, bits, choices, err := txhelpers.SSGenVoteChoices(msgTx, db.params)
+		if err != nil {
+			return &apitypes.ExplorerTxData{
+				Tx:       tx,
+				VinAddrs: prevOutAddresses,
+				Type:     txType,
+				Fee:      txFee,
+				FeeRate:  txFeeRate,
+			}
+		}
+		vinfo = &apitypes.VoteInfo{
+			Validation: apitypes.BlockValidation{
+				Hash:     validation.Hash.String(),
+				Height:   validation.Height,
+				Validity: validation.Validity,
+			},
+			Version: version,
+			Bits:    bits,
+			Choices: choices,
+		}
+	} else {
+		vinfo = nil
+	}
+
+	return &apitypes.ExplorerTxData{
+		Tx:       tx,
+		VinAddrs: prevOutAddresses,
+		Type:     txType,
+		Fee:      txFee,
+		FeeRate:  txFeeRate,
+		VoteInfo: vinfo,
+	}
+}
+
+// GetRawTransactionWithPrevOutAddresses looks up the previous outpoints for a
+// transaction and extracts a slice of addresses encoded by the pkScript for
+// each previous outpoint consumed by the transaction.
+func (db *wiredDB) GetRawTransactionWithPrevOutAddresses(txid string) (*apitypes.Tx, [][]string) {
+	tx, _ := db.getRawTransaction(txid)
+	if tx == nil {
+		return nil, nil
+	}
+
+	prevOutAddresses := make([][]string, len(tx.Vin))
+
+	for i := range tx.Vin {
+		vin := &tx.Vin[i]
+		if vin.IsCoinBase() /* || vin.IsStakeBase() */ {
```

Comment:
FYI, I left this comment because there is a PR for dcrd to introduce changes to `Vin` that will include a Stakebase field and similar method to check it.

Reply:
Noted